File: src/main/java/net/prosavage/bedfilevisualizer/bedfileclient/BedFileClient.java
package net.prosavage.bedfilevisualizer.bedfileclient;

import java.io.File;
import java.io.IOException;
import java.util.Arrays;
import java.util.List;

public class BedFileClient {

	private List<File> files;
	private boolean useMultiThreading;
	private Runtime runtime;

	public BedFileClient(List<File> files) {
		this.files = files;
		useMultiThreading = files.size() > 2;
		runtime = Runtime.getRuntime();
	}


	public File runIntersect(File outputFile) {
		try {
			ProcessBuilder processBuilder = new ProcessBuilder();
			processBuilder.directory(this.files.get(0).getParentFile());
			processBuilder.redirectOutput(outputFile);
			if (useMultiThreading) {
				List<String> command = Arrays.asList(
						  "python",
						  "/home/prosavage/Projects/School/HudsonAlphaTechChallenge/rum.py",
						  files.size() + ""
				);
				files.forEach((file -> command.add(file.getName())));
				processBuilder.start();
				return outputFile;
			}
			processBuilder.command("bedtools", "intersect", "-a", this.files.get(0).getAbsolutePath(), "-b", files.get(0).getAbsolutePath());
			processBuilder.start();

		} catch (IOException ex) {
			ex.printStackTrace();
		}
		return outputFile;
	}


}

File: src/main/java/net/prosavage/bedfilevisualizer/bedfileclient/BedFileClient.kt
package net.prosavage.bedfilevisualizer.bedfileclient

import java.io.File

class BedFileClient
/**
 * This runs the bed file client.
 *
 * @param files - files for the client to process
 */
{


        constructor(files: MutableList<File>) {
            this.files = files
            useMultiThreading = files.size > 2
        runtime = Runtime.getRuntime()
        }
    private var files: MutableList<File>? = null
    private var useMultiThreading: Boolean = false
    private var runtime: Runtime

    /**
     * Runs the intersect algorithm, there are no arguments.
     * This is the first part.
     */
    fun runIntersect(outputFile: File): File {
        if (useMultiThreading) {
            // TODO: Call python.
        }

        var processBuilder = ProcessBuilder()
        processBuilder.command("bedtools","intersect", "-a", this.files!![0].path, "-b" , files!![1].path)
        processBuilder.directory(this.files!![0].parentFile)
        processBuilder.redirectOutput(outputFile)
        processBuilder.start()
//        runtime.exec("bedtools intersect -a ${this.files!![0].absolutePath} -b ${files!![1].absolutePath}", null, File(files!![0].parent))
        System.out.println("done")
        return outputFile
    }

    /**
     * Runs the window algorithm.
     * This is the 2nd part.
     *
     * @param bp - the interval in base pairs
     */
    fun runWindow(bp: Int, outputFile: File): File {
        if (useMultiThreading) {
            //TODO: Call python.
        }
        var processBuilder = ProcessBuilder()
        processBuilder.command("bedtools","window", "-a", this.files!![0].path, "-b" , files!![1].path, "-w", bp.toString(), "output.bed")
        processBuilder.directory(this.files!![0].parentFile)
        processBuilder.redirectOutput(outputFile)
        processBuilder.start()
        return outputFile
    }
//N, K


}

